sdk/keyvault/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,141 @@
+# Troubleshooting Azure Key Vault SDK Issues
+
+The Azure Key Vault SDKs for .NET use a common HTTP pipeline and authentication to create, update, and delete secrets,
+keys, and certificates in Key Vault and Managed HSM. This troubleshooting guide contains mitigation steps for issues
+common to these SDKs.
+
+For any package-specific troubleshooting guides, visit any of the following:
+
+* [Troubleshooting Azure Key Vault Administration SDK Issues](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/keyvault/Azure.Security.KeyVault.Administration/TROUBLESHOOTING.md)
+* [Troubleshooting Azure Key Vault Certificates SDK Issues](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/keyvault/Azure.Security.KeyVault.Certificates/TROUBLESHOOTING.md)
+* [Troubleshooting Azure Key Vault Keys SDK Issues](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/keyvault/Azure.Security.KeyVault.Keys/TROUBLESHOOTING.md)
+* [Troubleshooting Azure Key Vault Secrets SDK Issues](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/keyvault/Azure.Security.KeyVault.Secrets/TROUBLESHOOTING.md)
+
+See [Azure SDK diagnostics](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/Diagnostics.md)
+for help diagnosing various problems across all our Azure SDKs for .NET.
+
+## Table of Contents
+
+* [Troubleshooting Authentication Issues](#troubleshooting-authentication-issues)
+  * [HTTP 401 Errors](#http-401-errors)
+    * [Frequent HTTP 401 Errors in Logs](#frequent-http-401-errors-in-logs)
+    * [AKV10032: Invalid issuer](#akv10032-invalid-issuer)
+    * [Other Authentication Issues](#other-authentication-issues)
+  * [HTTP 403 Errors](#http-403-errors)
+    * [Operation Not Permitted](#operation-not-permitted)
+    * [Access Denied to First Party Service](#access-denied-to-first-party-service)
+* [Other Service Errors](#other-service-errors)
+  * [HTTP 429: Too Many Request](#http-429-too-many-requests)
+
+## Troubleshooting Authentication Issues
+
+### HTTP 401 Errors
+
+HTTP 401 errors may indicate problems authenticating.
+
+#### Frequent HTTP 401 Errors in Logs
+
+Most often, this is expected. Azure Key Vault issues a challenge for initial requests that force authentication. You may
+see these errors most often during application startup, but may also see these periodically during the application's
+lifetime when authentication tokens are near expiration.
+
+If you are not seeing subsequent exceptions from the Key Vault SDKs, authentication challenges are likely the cause.
+
+#### AKV10032: Invalid issuer
+
+You may see an error similar to:
+
+```text
+Azure.RequestFailedException : Service request failed.
+Status: 401 (Unauthorized)
+
+Content:
+{"error":{"code":"Unauthorized","message":"AKV10032: Invalid issuer. Expected one of https://sts.windows.net/{tenant 1}/, found https://sts.windows.net/{tenant 2}/."}}
+```
+
+This is most often caused by being logged into a different tenant than the Key Vault authenticates.
```

Comment:
Is the tenant discovery feature relevant here?

Reply:
Probably, but I'd expect that to be detailed in the Identity TSG, right?

Reply:
I don't know, but we should mention it here too, in case they come at it from a Key Vault perspective.

Reply:
"tenant discovery" isn't really a feature of Azure.Identity, but rather a feature of clients which can determine the authority / tenant via an auth challange.

@christothes what should this mention about tenant discovery? Just that the latest package supports it and to upgrade? There's nothing the user can do to configure it.

Reply:
> "tenant discovery" isn't really a feature of Azure.Identity, but rather a feature of clients which can determine the authority / tenant via an auth challange.

I agree.

> @christothes what should this mention about tenant discovery? Just that the latest package supports it and to upgrade? There's nothing the user can do to configure it.

I think it makes the most sense for clients that support the feature to mention the version requirements. KV does this in its changelog. Snice there are so few that do, it may be misleading to mention it in Identity.